src/AddTodoForm.js

```diff
@@ -1,7 +1,12 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useRef } from "react";
 
 const AddTodoForm = ({ onAddTodo }) => { // accept onAddTodo as a prop
   const [todoTitle, setTodoTitle] = useState("");
+  const inputRef = useRef(null);
+
+  useEffect (() => {
+    inputRef.current.focus();
+  }, []);
```

Comment:
I'm sorry there was a glitch and all of my comments got deleted so bear with me

So it looks like you did the `useRef` logic in two places, and while you created the `InputWithLabel` component you haven't imported it here. It also feels like some of the deviations you've made the curriculum earlier on are fighting against you so I'm gonna try and get you back on track. This file should look more like this

```
import React, { useState } from "react";
import InputWithLabel from "./InputWithLabel";

export default function AddTodoForm({ onAddTodo }) {
  const [todoTitle, setTodoTitle] = useState("");

  //originally you doing this directly in the input like this: 
//onChange((e)=>setTodoTitle(e.target.value)) which is 
//totally valid, but now that things are moving around it 
//probably makes more sense to define and pass this 
//handler into InputWithLabel instead of the state setter.
  function handleTitleChange(e) {
    setTodoTitle(e.target.value);
  }

  function handleAddTodo(e) {
    e.preventDefault();
    onAddTodo({ title: todoTitle, id: Date.now() });
    setTodoTitle("");
  }

  return (
    <form onSubmit={handleAddTodo}>
      <InputWithLabel todoTitle={todoTitle} handleTitleChange={handleTitleChange}>
        Title //this is what we're passing in as children in the InputWithLabel component
      </InputWithLabel>
      <button type="submit">Add</button>
    </form>
  );
}

```

Reply:
got it! Thank you